mycpp/gc_str.cc

```diff
@@ -482,3 +485,96 @@ List<Str*>* Str::split(Str* sep) {
 
   return result;
 }
+
+// TODO: We should reduce the number of allocations and copies done here.
+static inline Str* _StrFromFormat(const char* fmt, va_list args) {
```

Comment:
To keep things consistent with python's `%` operator, I'm pretty liberal with assertions in this function. I could be convinced we should just return `nullptr` and have the callers check for failures. It may require changing a lot of call sites, though...